src/functions.php

```diff
@@ -62,3 +62,12 @@ function (ExceptionStack $exceptionStack) use ($failMessage) {
         }
     );
 }
+
+/**
+ * @template T
+ * @psalm-param Result<T> $result
+ */
+function success_or_fail(Result $result, string $failMessage)
```

Comment:
Difference from `extract_or_fail` is that `success_or_fail` doesn't return any value. It's kind of weird to write `extract_or_fail` client side without storing the returned value somewhere.